pkg/api/pod/util.go

```diff
@@ -591,6 +591,35 @@ func dropDisabledFields(
 	}
 
 	dropDisabledPodAffinityTermFields(podSpec, oldPodSpec)
+
+	dropDisabledTopologySpreadConstraintsFields(podSpec, oldPodSpec)
+}
+
+// dropDisabledTopologySpreadConstraintsFields removes disabled fields from PodSpec related
+// to TopologySpreadConstraints only if it is not already used by the old spec.
+func dropDisabledTopologySpreadConstraintsFields(podSpec, oldPodSpec *api.PodSpec) {
+	if !utilfeature.DefaultFeatureGate.Enabled(features.MinDomainsInPodTopologySpread) &&
+		!minDomainsInUse(oldPodSpec) &&
+		podSpec != nil {
+		for i := range podSpec.TopologySpreadConstraints {
```

Comment:
It's more appropriate to check the usage in each constraint independently.

Otherwise, if there is one constraint with the field used, one could add the field in other constraint, even if the feature is disabled.

The purpose of the strategy is that, if the feature is disabled, you only allow existing fields to remain set, but not to add new usages.

Reply:
correlating the constraints between the old and new objects is probably more trouble than it's worth... an object-wide permit/clear decision is probably fine

Reply:
I updated the implementation as @alculquicondor suggestion. 

@liggitt if you have any concerns about the implementation, let me know.

Reply:
Please follow @liggitt's advice instead.

Reply:
Sure.